spec/secure-hash.spec.js

```diff
@@ -0,0 +1,44 @@
+"use strict";
+
+describe("secureHash", ()  => {
+    var secureHash, crypto;
+
+    beforeEach(() => {
+        var base64, promiseMock;
+
+        crypto = require("crypto");
+        promiseMock = require("./mock/promise-mock");
+        base64 = require("../lib/base64");
+        secureHash = require("../lib/secure-hash")(base64, crypto, promiseMock);
+    });
+    it("hashes a passed in string", (done) => {
+        secureHash.hashAsync("rRTcBER_EiFUsRa34Hj5Zpok", {
+            algorithm: "sha256",
+            hashLength: 24,
+            iterations: 10000,
+            salt: "pinkFluffyUnicornsDancingOnRainbows"
+        }).then((result) => {
+            expect(result.toString("binary")).toBe("9GnOLZ_xAlfMA4C6DHsjNJJpsShI_TgR");
```

Comment:
Here, and elsewhere in the file, you convert a string to a string.  That is unnecessary.